spec/index.bs

```diff
@@ -7322,11 +7322,11 @@ dictionary GPURenderPassDepthStencilAttachment {
         is a [=combined depth-stencil format=]:
         - |this|.{{GPURenderPassDepthStencilAttachment/depthReadOnly}} must be equal to |this|.{{GPURenderPassDepthStencilAttachment/stencilReadOnly}}
     - If |this|.{{GPURenderPassDepthStencilAttachment/depthReadOnly}} is `true`:
-        - |this|.{{GPURenderPassDepthStencilAttachment/depthLoadOp}} must be {{GPULoadOp/"load"}}.
-        - |this|.{{GPURenderPassDepthStencilAttachment/depthStoreOp}} must be {{GPUStoreOp/"store"}}.
+        - |this|.{{GPURenderPassDepthStencilAttachment/depthLoadOp}} must not [=map/exist|be provided=].
+        - |this|.{{GPURenderPassDepthStencilAttachment/depthStoreOp}} must not [=map/exist|be provided=].
     - If |this|.{{GPURenderPassDepthStencilAttachment/stencilReadOnly}} is `true`:
-        - |this|.{{GPURenderPassDepthStencilAttachment/stencilLoadOp}} must be {{GPULoadOp/"load"}}.
-        - |this|.{{GPURenderPassDepthStencilAttachment/stencilStoreOp}} must be {{GPUStoreOp/"store"}}.
+        - |this|.{{GPURenderPassDepthStencilAttachment/stencilLoadOp}} must not [=map/exist|be provided=].
+        - |this|.{{GPURenderPassDepthStencilAttachment/stencilStoreOp}} must not [=map/exist|be provided=].
```

Comment:
Something to discuss: Should we also allow them to be set to "load" and "store"? I thought it made sense since that's the effective behavior, but OTOH maybe "store" on a read-only thing is too nonsensical.

Reply:
yeah, I don't think we should allow that

Reply:
I thought one of the points of this refactor was to get rid of the awkward requirement of setting "store" on a read-only attachment? I've updated the PR but have still required that load/store are undefined for read only attachments. Feels cleaner that way, happy to alter it though.

Reply:
Yeah, we wanted to not _require_ "store", but we could still _allow_ it.

----

The latest revision is more strict than before (and a breaking change) because now it disallows setting depthLoadOp when the attachment doesn't have a depth aspect. I don't think it's worth the breaking change to change that. So I guess that comes out to:

- If format has a depth aspect:
    - If depthReadOnly is false:

        - depthLoadOp/depthStoreOp must be provided

        otherwise:

        - depthLoadOp/depthStoreOp must not be provided

? Specifically this ignores depthReadOnly/depthLoadOp/depthStoreOp entirely if there's no depth aspect.

Reply:
That feels weird, honestly. If I don't have something X, and I'm telling WebGPU to do operation A on X, that should be an error, not ignored.

Reply:
If there's no depth aspect, could we still feasibly support `"clear"`+`"discard"`? Seems like that _might_ work on tilers but fail on other architectures.

In any case, if that's something that can/should be supported then I can see allowing a depth (and stencil) ops when no aspect is present, but otherwise letting developers set those values seems like we're setting up developer expectations of which values will be available in the shaders that can't be fulfilled.

Reply:
The depth/stencil attachment format is probably used inside the render pass too (even on tilers). E.g. I would expect the tile's depth buffer to actually be 16/24/32-bit according to the format. So if there's no allocated stencil aspect I wouldn't expect it to work. In any case, I am pretty sure none of the APIs allow that.

Maybe we should just have a breaking change here then. Agree it does seem a bit cleaner.

Reply:
If it'll be breaking should we surface it to the group first?

Reply:
> If there's no depth aspect, could we still feasibly support "clear"+"discard"? Seems like that might work on tilers but fail on other architectures.

That sounds like a memory-less (or transient) render target. It's a different feature, and I don't think we discussed it much. Let's not be distracted by it.